### PR TITLE
Add TypeScript support, add package-lock file

### LINKED ISSUE #6: No support for typescript
When you try to import this project in a typescript codebase you get the following error:

Could not find a declaration file for module 'format-currency'. '/node_modules/format-currency/index.js' implicitly has an 'any' type.
  Try `npm install @types/format-currency` if it exists or add a new declaration (.d.ts) file containing `declare module 'format-currency';`

And a definition files does not exist in the Definitely Typed project, but it's always the better choice that the project itself provides it's own definitions

### PR DESCRIPTION
**Pull request by sandman21dan**
Hello!

Liked this minimalist project to format some currencies, I like to work in TypeScript and this project is missing types, this PR fixes #6 and also adds a package-lock.json file for faster installs and better dependency versioning for npm versions >=5

hope you can merge and release this!

### CODE REVIEW

**Comment by jpike88** [comment]
@RyanZim are you going to approve this??

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  localeMatcher?: string;
+  useGrouping?: boolean;
+  minimumIntegerDigits?: number;
+  minimumFractionDigits?: number;
+  maximumFractionDigits?: number;
+  minimumSignificantDigits?: number;
+  maximumSignificantDigits?: number;
+}
+
+export default function formatCurrency(number: number | string | null | undefined, opts?: FormatCurrencyOptions): string;
```
Comment (rohmanhm) [inline index.d.ts:13]: I think `export default` will not work with this package.
https://github.com/ExodusMovement/format-num/blob/ebf9ab8890a8d9cab780bfba6675cf17d5467904/index.js#L67

I will suggest something like this

```ts
function formatCurrency(number: number | string | null | undefined, opts?: FormatCurrencyOptions): string;

export = formatCurrency
```

**Comment by sandman21dan** [comment]
Bump

**Comment by jpike88** [comment]
If @RyanZim doesn't sort this out soon, I'll release this on npm myself with the PR applied.

**Comment by jprichardson** [comment]
I'm okay with us supporting this. Please remove `package-lock.json`, `package.json` modifications, and remove `CHANGELOG.md` modifications.

**Comment by RyanZim** [comment]
Closing for lack of updates.

**Pull request closed without merging**